ngraph/src/ngraph/frontend/onnx_import/op/quantize_linear.cpp

```diff
@@ -16,62 +16,238 @@
+                                        const element::Type& data_type)
+                    {
+                        std::shared_ptr<ngraph::Node> input_low;
+                        std::shared_ptr<ngraph::Node> input_high;
+                        const auto& zero_point =
+                            std::make_shared<default_opset::Convert>(y_zero_point, data_type);
+
+                        input_low = std::make_shared<default_opset::Multiply>(
+                            y_scale,
+                            std::make_shared<default_opset::Subtract>(output_low, zero_point));
+                        input_high = std::make_shared<default_opset::Multiply>(
+                            y_scale,
+                            std::make_shared<default_opset::Subtract>(output_high, zero_point));
+
+                        return std::make_tuple(input_low, input_high);
                     }
 
-                    Shape y_scale_shape = y_scale->get_shape();
-                    Shape y_zero_point_shape = y_zero_point->get_shape();
+                    std::shared_ptr<ngraph::Node>
+                        make_fake_quantize(const std::shared_ptr<ngraph::Node>& y_scale,
+                                           const std::shared_ptr<ngraph::Node>& y_zero_point,
+                                           const std::shared_ptr<ngraph::Node>& data)
+                    {
+                        const element::Type& destination_type = y_zero_point->get_element_type();
+                        const element::Type& data_type = data->get_element_type();
+
+                        std::shared_ptr<ngraph::Node> output_low;
+                        std::shared_ptr<ngraph::Node> output_high;
+                        std::tie(output_low, output_high) =
+                            detail::get_output_bands(destination_type, data_type);
+
+                        std::shared_ptr<ngraph::Node> input_low;
+                        std::shared_ptr<ngraph::Node> input_high;
+                        std::tie(input_low, input_high) = detail::get_input_bands(
+                            y_scale, y_zero_point, output_low, output_high, data_type);
+
+                        const std::size_t levels = 1 << destination_type.bitwidth();
 
-                    return {std::make_shared<ngraph::opset0::Quantize>(
-                        x,
-                        y_scale,
-                        y_zero_point,
-                        y_zero_point->get_element_type(),
-                        axes,
-                        ngraph::opset0::Quantize::RoundMode::ROUND_NEAREST_TOWARD_EVEN)};
+                        return std::make_shared<default_opset::Convert>(
+                            std::make_shared<default_opset::FakeQuantize>(
+                                data, input_low, input_high, output_low, output_high, levels),
+                            destination_type);
+                    }
                 }
+            }
+
+            namespace set_1
```

Comment:
Why is it set_1 but not set_10? Operation was introduced in ONNX opset 10.

Reply:
@slyalin We're following robustness principle here. Moreover as far as I remember at the beginning of work on ONNX Importer we were encountering models which had operators not yet introduced in the opset set in the protobuf model. In general all operators _first_ version in ONNX importer's is `set_1` (as long as we can support the op we do). All later spec modifications are updated in set version which is in sync with ONNX opset version.